### ISSUE TITLE
Port twisted.test.test_defgen and twisted.internet.test.test_inlinecb to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6010]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6010|
|Type|enhancement|
|Created|2012-09-21 17:34:43Z|
|Branch|https://github.com/twisted/twisted/tree/defgen-py3-6010|

`twisted.test.test_defgen` and `twisted.internet.test.test_inlinecb` test inlineCallbacks and deferredGenerator. It should be ported to Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__6010 6010
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_defgen_py3_6010 branches/defgen-py3-6010
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1348248883000000 1348248883000000
changetime__1349456860000000 1349456860000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

How about not bothering with `test_defgen` or `deferredGenerator`?  I just filed [#6044](https://github.com/twisted/twisted/issues/6044).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst commented|
|-|-|

There are a bunch of ported tests that rely on deferredGenerator, so probably best to port it (unless it turns out to be easier to fix those tests). In particular, I suspect it will be trivial to do.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/defgen-py3-6010 started.

I renamed a hidden module, removed an exec (both unnecessary now that we use modern Pythons), and fixed a couple of syntax errors in exception catching; the tests are otherwise unmodified.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst commented|
|-|-|

(In [35945]) Branching to 'defgen-py3-6010'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @itamarst|
|-|-|

Thanks.

The ticket reference at the top of test_inlinecb.py doesn't seem too useful anymore?  At least, these tests are no longer skipped in any way.  Actually, the tests in test_defgen.py aren't skipped anymore either and there is no `generator_failure_tests.py` file in `twisted/test/` at all.  Does that mean [#4182](https://github.com/twisted/twisted/issues/4182) is resolved?

No review comments about the branch itself.  Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst set status to `closed`|
|-|-|

(In [35976]) Merge defgen-py3-6010.

Author: itamar
Review: exarkun
Fixes: [#6010](https://github.com/twisted/twisted/issues/6010)

Port deferredGenerator and inlineCallbacks tests to Python 3.
